dietpi/dietpi-software

```diff
@@ -10807,7 +10807,18 @@ _EOF_
 			fi
 
 			# ARM: Install known required Python modules via APT to avoid time-consuming compiling and large build dependencies (on x86 pre-compiled wheels are available)
-			(( $G_HW_ARCH == 10 )) || G_AGI python3-{lxml,numpy}
+			if (( $G_HW_ARCH != 10 ))
+			then
+				DEPS_LIST='python3-numpy python3-lxml'
+				if (( $G_HW_ARCH < 3 ))
+				then
+					# ARMv6/7 add libatlas3-base as dependency
+					DEPS_LIST+=' libatlas3-base'
+				fi
+			fi
+			# shellcheck disable=SC2086
+			G_AGI $DEPS_LIST
+			unset -v DEPS_LIST
```

Comment:
```suggestion
			local apackages=()
			(( $G_HW_ARCH == 10 )) || apackages=('python3-lxml' 'python3-numpy')
			(( $G_HW_ARCH < 3 )) && apackages+=('libatlas3-base')
			G_AGI "${apackages[@]}"
```

Reply:
Regarding missing numpy wheels on piwheels: https://piwheels.org/project/numpy/
numpy v1.22 simply requires at last Python 3.8, hence cannot be used on Buster, and numpy v1.19 requires at least Python 3.6, hence cannot be used on Stretch. `pip` hence is able to download a pre-compiled wheel of the most recent installable version, i.e. build essential dependency and those two Python 3 packages shouldn't be required anymore in any case. I'll test this.

Reply:
Latest Python 3.7 compatible numpy v1.21.5 is currently missing on piwheels: https://github.com/piwheels/packages/issues/287

Hence `g++` is required to compile it. Let's see whether piwheels can add it soon, otherwise we'll leave build essential dependency in place for ARMv6/7.

Reply:
> we'll leave build essential dependency in place for ARMv6/7
On Buster only?

Reply:
Ah yes, only on Buster.

Reply:
Still failing on ARMv6/7 Buster. Looks like the fortran compiler is additionally required 😄. Doesn't sound reasonable anymore, I think we should count on piwheels to provide the latest Python 3.7 compatible numpy and skip trying to make it work without. I'll however continue trying it tomorrow.

Reply:
Workaround: https://github.com/piwheels/packages/issues/287#issuecomment-1036500818